.github/workflows/call_test_package.yml

```diff
@@ -0,0 +1,142 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Tests
+
+on:
+  # Allows you to run this workflow from another workflow
+  workflow_call:
+
+# permission for reporting
+permissions:
+  contents: read
+  actions: read
+  checks: write
+  pull-requests: write
+
+
+jobs:
+  build-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true  # if on job fail all the other jobs are cancelled
+      matrix:
+        python-version: ["3.13", "3.12"] #CHECK: not forget to add new version of python
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    
+    # name for files
+    env:
+      REPORT_NAME: ${{ matrix.python-version }}-os-${{ matrix.os }}
+
+    steps:
+    # Checkout to the branch for getting the ritgh code
+    - uses: actions/checkout@v4
+    # Install python
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    # Install dependency for running the tests
+    - name: Install the dependance
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install .[test]
+    # Run the tests with coverage and report add this end
+    # option for the coverage: --cov=. --cov-report=xml:... 
+    # option for mardown report: --md-report --md-report-output= ...
+    # option for html report: --self-contained-html --html=report-...
+    # option for duration report: --pytest-durations= number of second --pytest-resultlog=log-...
+    - name: Generate coverage report
+      run: |
+        python -m pytest\
+         --cov=. --cov-report=xml:cov/cov-${{env.REPORT_NAME}}.xml\
+         --md-report --md-report-output=test-results-${{env.REPORT_NAME}}.md\
+         --self-contained-html --html=report-${{env.REPORT_NAME}}.html\
+         --pytest-durations=10 --pytest-resultlog=log-${{env.REPORT_NAME}}.txt
```

Comment:
see if you can move some of those to the pytest config in pyproject.toml:
- this will make sure they run all the time
- this will make the CI set up easier to read

or maybe keep them there for now and then move them your tox command if / when you set up tox

Reply:
I added these options only here because there are the generation of reports which I don't need when I run the test on my computer.